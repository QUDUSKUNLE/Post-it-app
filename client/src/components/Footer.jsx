import React from 'react';

/**
 * @export
 * @description - renders Footer Component
 * @class Footer
 * @extends {React.Component}
 */
export default class Footer extends React.Component {
  render() {
    return (
      <div className="footer">
        <div className="container">
          <center>
           Andela 2017
          </center>
        </div>
      </div>
    );
  }
}
