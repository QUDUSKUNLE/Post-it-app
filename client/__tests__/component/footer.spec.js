import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import Footer from '../../src/components/Footer';

describe('PostIt-app', () => {
  it('expects Footer component to be defined', () => {
    expect(Footer).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<Footer/>);
    expect(component).toMatchSnapshot();
    expect(component.find('.footer')).toHaveLength(1);
    expect(component.find('div')).toHaveLength(1);
  });
});
