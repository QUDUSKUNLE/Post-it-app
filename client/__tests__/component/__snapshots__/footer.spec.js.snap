// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostIt-app should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="footer"
  >
    <div
      className="container"
    >
      <center>
        Andela 2017
      </center>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="footer"
    >
      <div
        className="container"
      >
        <center>
          Andela 2017
        </center>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Footer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Footer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="footer"
        >
          <div
            className="container"
          >
            <center>
              Andela 2017
            </center>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="footer"
        >
          <div
            className="container"
          >
            <center>
              Andela 2017
            </center>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Footer />,
}
`;
